shell/src/main/java/alluxio/cli/docgen/ConfigurationDocGenerator.java

```diff
@@ -197,26 +241,72 @@ public static void writeYMLFile(Collection<? extends PropertyKey> defaultKeys, S
     } finally {
       try {
         closer.close();
+        if (validate) {
+          compareFiles(YML_FILE_NAMES, filePath, "YML");
+        }
       } catch (IOException e) {
         LOG.error("Error while flushing/closing YML files for description of Property Keys "
             + "FileWriter", e);
       }
+      for (String fileName : YML_FILE_NAMES) {
+        String fileNameTemp = TEMP_PREFIX.concat(fileName);
+        File tempFile = new File(PathUtils.concatPath(filePath , fileNameTemp));
+        if (tempFile.exists()) {
+          tempFile.delete();
+        }
+      }
+    }
+  }
+
+  /**
+   * Helper method that compare the temp file and the committed file when validate flag is used.
+   *
+   * @param fileNames the name of the file
+   * @param filePath the path to the file
+   * @param fileType the type of the file that we are comparing (CSV/YML)
+   */
+  public static void compareFiles(String[] fileNames, String filePath, String fileType)
+      throws IOException {
+    boolean hasDiff = false;
+
+    for (String fileName : fileNames) {
+      String fileNameTemp = TEMP_PREFIX.concat(fileName);
+      File committedFile = new File(filePath, fileName);
+      File tempFile = new File(filePath, fileNameTemp);
+      if (!committedFile.exists()) {
+        System.out.println("Committed file does not exists");
+        continue;
+      }
+      if (!tempFile.exists()) {
+        System.out.println("Temporary generated file does not exists");
+        continue;
+      }
+      if (!FileUtils.contentEquals(committedFile,
+          tempFile)) {
+        hasDiff = true;
+        System.out.println("Config file " + fileName + " changed.");
```

Comment:
use printf instead of println to take advantage of string substitution. add `%n` to append a newline at the end to mimic what println would have done
`System.out.println("Config file %s changed.%n", fileName);`

and same for all other `System.out.println`s that concats a string variable value